test/test_tables.jl

```diff
@@ -89,5 +80,299 @@ using Tables
+                        end
+                        if withmissingfield
+                            AG.addfeature(newlayer) do newfeature
+                                # No Id field set
+                                AG.setfield!(newfeature, name_idx, data[geomfamily].emptyfield.name)
+                                AG.setgeom!(newfeature, 0, data[geomfamily].emptyfield.geom)
+                            end
+                        end
+                    end
+                end
+
+                if drvshortname == "GML"
+                    return AG.read(filename, 
+                        options=[
+                        "EXPOSE_GML_ID=NO"
+                        "EXPOSE_FID=NO"
+                        ])
+                else
+                    return AG.read(filename)
+                end
+            end
+
+            """
+                clean_test_files()
+
+            Cleans test files generated by `get_test_dataset`
+            """
+            function clean_test_dataset_files()
+                ds_file_extension = Dict(
+                    "ESRI Shapefile" => "",
+                    "GeoJSON" => ".geojson",
+                    "CSV" => ".csv",
+                    "GML" => ".gml",
+                    "KML" => ".kml",
+                    "GPKG" => ".gpkg",
+                    "FlatGeobuf" => ".fgb",
+                )
+                filename_root = "test_dataset"
+
+                # Clean previously created files for new dataset
+                for (drvshortname, file_extension) in ds_file_extension
+                    isfile(joinpath(@__DIR__, "tmp", filename_root * file_extension)) && rm(joinpath(@__DIR__, "tmp", filename_root * file_extension))
+                    isfile(joinpath(@__DIR__, "tmp", filename_root * ".xsd")) && rm(joinpath(@__DIR__, "tmp", filename_root * ".xsd"))
+                    isfile(joinpath(@__DIR__, "tmp", filename_root * ".tmp")) && rm(joinpath(@__DIR__, "tmp", filename_root * ".tmp"))
+                    isdir(joinpath(@__DIR__, "tmp", filename_root)) && rm(joinpath(@__DIR__, "tmp", filename_root), recursive=true)
+                end
+            end
+
+            toWKT_withmissings = (x -> ismissing(x) ? missing : AG.toWKT(x))
+            columntablevalues_toWKT = (x -> (toWKT_withmissings.(x[1]), x[2], x[3]))
+            tupleoftuples_equal = ((x, y) -> length(x) ==length(y) && all([all(x[i] .=== y[i]) for i in 1:length(x)]))
+            
+            """
+                test_layer_to_table(
+                    drvshortname::String, 
+                    geomfamilly::String,
+                    withmissinggeom::Bool,
+                    withmissingfield::Bool,
+                    withmixedgeomtypes::Bool,
+                    reference_geotable::Tuple;
+                    testing::Bool=true)
+
+            Creates a dataset from scratch with `get_test_dataset` and converts it to a columntable
+
+            If `testing = false`: returns a `NamedTuple` with:
```

Comment:
The `testing = false` option was intended as a convenience to further add some tests with different options on driver, geometry family, mixed or not mixed geometry types, and various missing values.

I can drop it or add a comment for its purpose.

Waiting for your preference, I have committed a comment to the function

Reply:
Ah, in that case my slight preference might be to leave out the `testing=false` part because of https://docs.julialang.org/en/v1/manual/performance-tips/#Write-%22type-stable%22-functions and also because the function name starts with `test_` which makes it feel more intuitive for it to return a boolean result.

If it's helpful as a convenience function, we can create a separate (differently-named) function that returns the named tuple.

Reply:
Ok I have created a separate function `layer_to_columntable_with_WKT`